features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -93,17 +95,51 @@ internal class Base64Serializer private constructor(
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
         base64SerializerCallback: Base64SerializerCallback?
     ) {
-        val base64String = convertBmpToBase64(drawable, bitmap, shouldCacheBitmap)
-        finalizeRecordedDataItem(base64String, imageWireframe, base64SerializerCallback)
+        val byteArray = compressBitmap(bitmap)
+        val base64String = convertBmpToBase64(byteArray, drawable, bitmap, shouldCacheBitmap)
+        val hash = calculateHash(drawable, byteArray)
+        finalizeRecordedDataItem(base64String, hash, imageWireframe, base64SerializerCallback)
+    }
+
+    private fun calculateHash(
+        drawable: Drawable,
+        byteArray: ByteArray
+    ): String {
+        val md5Generator = MD5HashGenerator(logger)
+        val hashValue = md5Generator.generate(byteArray)
+
+        if (hashValue.isNotEmpty()) {
+            hashLRUCache?.put(drawable, hashValue)
+        }
+
+        return hashValue
     }
 
     @MainThread
-    private fun registerCacheForCallbacks(applicationContext: Context) {
-        if (isCacheRegisteredForCallbacks) return
+    private fun registerBase64LruCacheForCallbacks(applicationContext: Context) {
+        if (isBase64CacheRegisteredForCallbacks) return
 
         if (base64LRUCache is ComponentCallbacks2) {
             applicationContext.registerComponentCallbacks(base64LRUCache)
-            isCacheRegisteredForCallbacks = true
+            isBase64CacheRegisteredForCallbacks = true
+        } else {
+            // Temporarily use UNBOUND logger
```

Comment:
seems this comment is wrong, there is no `UNBOUND` logger here, and there is no need for `TODO` item.